CMakeLists.txt

```diff
@@ -538,89 +506,71 @@ elseif(MSVC)
 		/wd4503)
 endif()
 
-if (shared)
-	add_library(torrent-rasterbar SHARED ${sources2})
+if(static_runtime)
+	include(ucm_flags)
+	ucm_set_runtime(STATIC)
+	set(Boost_USE_MULTITHREADED ON)
+	set(Boost_USE_STATIC_RUNTIME ON)
+	set(OPENSSL_USE_STATIC_LIBS TRUE)
+	set(OPENSSL_MSVC_STATIC_RT TRUE)
+endif()
+
+if (NOT BUILD_SHARED_LIBS)
+	set(Boost_USE_STATIC_LIBS ON)
+endif()
+
+add_library(torrent-rasterbar
+	src/$<JOIN:${sources}, src/>
+	include/libtorrent/$<JOIN:${libtorrent_include_files}, include/libtorrent/>
+	include/libtorrent/extensions/$<JOIN:${libtorrent_extensions_include_files}, include/libtorrent/extensions/>
+	include/libtorrent/aux_/$<JOIN:${libtorrent_aux_include_files}, include/libtorrent/aux_/>
+)
+
+if (BUILD_SHARED_LIBS)
 	target_compile_definitions(torrent-rasterbar
 		PRIVATE TORRENT_BUILDING_SHARED
 		INTERFACE TORRENT_LINKING_SHARED
 	)
-	set_target_properties(torrent-rasterbar
-		PROPERTIES
-			CXX_VISIBILITY_PRESET "hidden"
-			VISIBILITY_INLINES_HIDDEN "true"
-			VERSION ${PROJECT_VERSION}
-			SOVERSION ${SOVERSION}
-	)
+endif()
+
+set_target_properties(torrent-rasterbar
+	PROPERTIES
+		CXX_VISIBILITY_PRESET "hidden"
+		VISIBILITY_INLINES_HIDDEN "true"
+		VERSION ${PROJECT_VERSION}
+		SOVERSION ${SOVERSION}
+)
+
+if (cxx_std_14 IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+	target_compile_features(torrent-rasterbar PUBLIC cxx_std_14)
+	message(STATUS "Building in C++14 mode")
```

Comment:
Is there any benefit to building as C++14? I don't recall if libtorrent enables any code based on C++ version. Unless there's a good reason to do this I'd rather stick with C++11 so that we catch any accidental use of C++14 or latter features.

Reply:
yes, there is, in `entry.hpp`, to use `strview_less`

Reply:
Moreover, it is required to use the latest available standard, because tests in libtorrent headers rely on the `__cplusplus` macro version, and when you build a client with a newer standard, libtorrent declarations follow that, so must do implementations.